pkg/types/validation/installconfig.go

```diff
@@ -555,3 +574,18 @@ func validateCloudCredentialsMode(mode types.CredentialsMode, fldPath *field.Pat
 	}
 	return allErrs
 }
+
+// validateURI checks if the given url is of the right format. It also checks if the scheme of the uri
+// provided is within the list of accepted schema provided as part of the input.
+func validateURI(uri string, fldPath *field.Path, proxyType string, schemes []string) field.ErrorList {
+	parsed, err := url.ParseRequestURI(uri)
+	if err != nil {
+		return field.ErrorList{field.Invalid(fldPath, uri, err.Error())}
+	}
+	for _, scheme := range schemes {
+		if scheme == parsed.Scheme {
+			return nil
+		}
+	}
+	return field.ErrorList{field.Invalid(fldPath, parsed.Scheme, fmt.Sprintf("%q is an invalid scheme for %s %s", parsed.Scheme, proxyType, uri))}
```

Comment:
Use `NotSupported` when there is a list of valid options.
```suggestion
	return field.ErrorList{field.NotSupported(fldPath, parsed.Scheme, schemes)}
```
Note, that this also means you do not need the `proxyType` parameter any more.